qt/python/mantidqt/widgets/superplot/SuperplotPresenter.py

```diff
@@ -0,0 +1,488 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from .SuperplotView import SuperplotView
+from .SuperplotModel import SuperplotModel
+
+from mantid.api import mtd
+from mantid.plots.axesfunctions import plot
+from mantid.plots.utility import MantidAxType
+
+
+class SuperplotPresenter:
+
+    BIN_MODE_TEXT = "Bin"
+    SPECTRUM_MODE_TEXT = "Spectrum"
+    _view = None
+    _model = None
+    _canvas = None
+
+    def __init__(self, canvas, parent=None):
+        self._view = SuperplotView(self, parent)
+        self._model = SuperplotModel()
+        self._canvas = canvas
+        self.parent = parent
+
+        if self.parent:
+            self.parent.plot_updated.connect(self.onPlotUpdated)
+
+        self._model.workspaceDeleted.connect(self.onWorkspaceDeleted)
+        self._model.workspaceRenamed.connect(self.onWorkspaceRenamed)
+        self._model.workspaceReplaced.connect(self.onWorkspaceReplaced)
+
+        #initial state
+        self._syncWithCurrentPlot()
+
+        self._updateList()
+        self._updateSpectrumSlider()
+        self._updatePlot()
+
+    def getSideView(self):
+        return self._view.getSideWidget()
+
+    def getBottomView(self):
+        return self._view.getBottomWidget()
+
+    def close(self):
+        if self.parent:
+            try:
+                self.parent.plot_updated.disconnect()
+            except:
+                pass
+        self._view.close()
+        del self._model
+
+    def _syncWithCurrentPlot(self):
+        """
+        This methods synchronize the model with the current plotted data. It
+        first checks that the plotted data are consistent (i.e. only bins, only
+        spectra), if not, it returns without updating the model.
+        """
+        figure = self._canvas.figure
+        axes = figure.gca()
+        artists = axes.get_tracked_artists()
+        if not artists:
+            self._view.setAvailableModes([self.SPECTRUM_MODE_TEXT,
+                                          self.BIN_MODE_TEXT])
+        else:
+            try:
+                args = axes.creation_args
+            except:
+                args = [{}]
+            if "axis" in args[0]:
```

Comment:
You mean the search for the axis key? Here I am checking that all curves are either using `MantidAxType.BIN` or `MantidAxType.SPECTRUM` axis. From what I understood of these `creation_args` is that if axis is not present, `MantidAxType.SPECTRUM` is used. Moreover, the axis type is a python enum and sometime the value is used instead of the name (`MantidAxType.SPECTRUM.value`, `MantidAxType.BIN.value`).